compiler/src/dotty/tools/dotc/core/TypeErasure.scala

```diff
@@ -605,6 +605,8 @@ class TypeErasure(isJava: Boolean, semiEraseVCs: Boolean, isConstructor: Boolean
       case tp: TypeVar =>
         val inst = tp.instanceOpt
         if (inst.exists) sigName(inst) else tpnme.Uninstantiated
+      case RefinedType(parent, nme.apply, refinedInfo) if parent.typeSymbol eq defn.PolyFunctionClass =>
+        sigName(defn.FunctionType(refinedInfo.resultType.paramNamess.head.length))
```

Comment:
It still seems weird that an explicit case is needed here, given that the default case below does `val erasedTp = this(tp) ... sigName(erasedTp)` which should end up going through https://github.com/lampepfl/dotty/blob/93d3c2a719a4d59c5c6e24028fc52146097e5b1b/compiler/src/dotty/tools/dotc/core/TypeErasure.scala#L450

Reply:
It doesn't get to the default case because It gets caught by the `TypeProxy` case immediately below.

Reply:
Aha, I see. But then you should be able to do something like `sigName(this(tp))` in this case, and add a comment mentioning this is only needed to avoid going through the TypeProxy case below.

Reply:
Done.